### PR TITLE
Move to config gem pattern.

### LINKED ISSUE #48: Move solr configurations away from Solr.yml to config gem
This will require a major version bump.

### PR DESCRIPTION
**Pull request by mejackreed**
This will require a major version bump.. targeted to v4.x

Closes #48 

I'm unsure why we were modifying our config values here: https://github.com/sul-dlss/base_indexer/commit/7962fefb9c8fc04aad33a42b1b3828015ab66c08#diff-913540fc5d53a19817eec9ea6a88d743R22

I can understand the need to standardize but it feels strange to me replacing what is in the configuration.

### CODE REVIEW

**Comment by drh-stanford** [comment]
I'm assuming that we need to make modifications to shared configs. Do we have open PRs for that?

**Comment by mejackreed** [comment]
This is for the gem, we would then need to add PR's to sw-indexer + shared_configs to use this (after a release)

File: lib/tasks/index.rake [hunk 1]
```diff
@@ -30,7 +30,7 @@ task :log_indexer => :environment do |t, args|
   raise 'Log type must be preassembly, remediate or csv.' unless ['preassembly','remediate','csv'].include? log_type
   raise 'Log file not found.' unless File.readable? log_file_path
 
-  target_config=BaseIndexer.solr_configuration_class_name.constantize.instance.get_configuration_hash[target.upcase]
+  target_config = Settings.SOLR_TARGETS[target]
```
Comment (drh-stanford) [inline lib/tasks/index.rake]: why are we removing the `.upcase` in this PR?

Comment (mejackreed) [inline lib/tasks/index.rake]: My thought here is that there is no way to pass to a non-upcased configuration. The configuration should be the configuration. If we want all upcased targets, can't we just upcase them ourselves in the config?

Comment (drh-stanford) [inline lib/tasks/index.rake]: That's fine with me. Just need a note for that in the release notes.

**Pull request merged**